### DISCUSSION TITLE
Indicate `value transformation` and `unit transformation` in adapters

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#1513] [Ideas]
StreamPipes allows users to configure simple value transformations or measurement conversions directly during adapter creation which are then applied directly when the data is read from the source:

<img width="1398" alt="Bildschirmfoto 2023-04-19 um 16 48 04" src="https://user-images.githubusercontent.com/50115603/233114972-2866ea69-8666-45f1-aa74-892b435a03cb.png">

This is a convenient feature which provides a shorthand for using a dedicated processing element in the pipeline definition.
Nevertheless, once the applied transformations are not apparent anymore, so it is somehow happening secretly.
I think this is not ideal and can lead to confusion if multiple users work with the same assets in StreamPipes.
Therefore, my suggestion would be add some information about this in the adapter info:
![adapter](https://user-images.githubusercontent.com/50115603/233115958-e77f07f3-32d4-4b21-a8ed-cf495a1f5eaf.gif)

and add a small indicator in the pipeline editor which then provides detailed information when hovering over it.

How do you think about this idea? Do you have any other places in mind where this information could be useful?
Or is it already somewhere accessible and I've missed it?

**Comment by dominikriemer** [comment]
I like this idea very much - this would add more transparency for users when working with adapters!

**Reply by bossenti** [reply]
Cool, I'll create an issue out of it!

**Reply by bossenti** [reply]
see https://github.com/apache/streampipes/issues/1534
